inc/class-s3-media-sync.php

```diff
@@ -34,10 +38,16 @@ public function get_settings_handler() {
 	}
 
 	public function get_s3_bucket() {
+		if ( $this->bucket === null ) {
```

Comment:
Returning an empty string here changes the return type contract of `get_s3_bucket()` from bucket name to `string|""`. Prefer returning `null` and updating call sites (with strict checks), or keep returning a string and always compute it from settings instead of `$this->bucket`. Suggested change:

```php
public function get_s3_bucket() {
    return $this->bucket ? $this->bucket->get_name() : null;
}
```

And annotate the method (docblock) to reflect `?string`.